### DISCUSSION TITLE
Is there a way to use CPU for set_asr_model()?

### DISCUSSION THREAD

**Discussion by vitotitto** [NVIDIA-NeMo/NeMo#3963] [Q&A] [answered]
I have issues with CUDA, so try to run models using CPU. 

As was previously advised, when loading from_pretrained  I use `map_location = 'cpu'` 

Is there a way to use cpu for this part? 

`asr_model = asr_ts_decoder.set_asr_model()
`

I am trying to follow the tutorial here 
https://github.com/NVIDIA/NeMo/blob/main/tutorials/speaker_tasks/ASR_with_SpeakerDiarization.ipynb

**Comment by tango4j** [comment, marked as answer]
Hi, Yes, you can do ASR inference on CPU
(1) Either it is restore_from or from_pretrained, feed the following argument
map_location=torch.device("cpu")

Go to this line to change this for ASR with Speaker Diarization
https://github.com/NVIDIA/NeMo/blob/1d9ee1c6d0b78efa4cc28609bfd57ecad36cbec6/nemo/collections/asr/parts/utils/decoder_timestamps_utils.py#L368

(2)please make sure you mask GPU by setting this:
export CUDA_VISIBLE_DEVICES=""; python offline_diarization_with_asr.py

If you don't mask (not visible by the system) GPU, the model will be operated on GPU although you specified cpu. This is because we are using dataloader based on gpu/cpu determined by torch.cuda.is_available.
